src/components/DashboardsBar/CreateDashboardButton/ChooseDashboardTypeModal.js

```diff
@@ -0,0 +1,106 @@
+import { useConfig } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useCallback, useState } from 'react'
+import { DashboardTypeRadio } from './DashboardTypeRadio.js'
+import { IconDashboardExternal } from './IconDashboardExternal.js'
+import { IconDashboardInternal } from './IconDashboardInternal.js'
+import styles from './styles/ChooseDashboardTypeModal.module.css'
+
+const TYPE_INTERNAL = 'INTERNAL'
+const TYPE_SUPERSET = 'SUPERSET'
+
+const autoFocus = (element) => {
+    element?.focus()
+}
+
+export const ChooseDashboardTypeModal = ({
+    onCancel,
+    onSelectSuperset,
+    onSelectInternal,
+}) => {
+    const {
+        systemInfo: { systemName },
+    } = useConfig()
+    const [selectedType, setSelectedType] = useState(TYPE_INTERNAL)
+    const handleDashboardTypeChange = useCallback((event) => {
+        setSelectedType(event.target.value)
+    }, [])
+    const isInternal = selectedType === TYPE_INTERNAL
+
+    return (
+        <Modal>
+            <form onSubmit={isInternal ? onSelectInternal : onSelectSuperset}>
+                <ModalTitle>
+                    {i18n.t('New dashboard: choose type', {
+                        nsSeparator: '###',
+                    })}
+                </ModalTitle>
+                <ModalContent>
+                    <fieldset
+                        ref={autoFocus}
+                        className={styles.dashboardTypeRadioGroup}
+                        tabIndex={0}
```

Comment:
This is triggering a SonarQube failure.

Reply:
We've discussed that setting focus on the first input on each modal would be a suitable solution to address both the SonarCube failure and maintaining a good UX. See https://github.com/dhis2/dashboard-app/pull/3185/commits/89b5f2cfc1847fdc71dc227f14a679ccb8436a9b and https://github.com/dhis2/dashboard-app/pull/3185/commits/e4adf3575afacd6a0dc1f58aced23167ef27f90b